app/src/main/java/com/arielwang/workoutlogger/features/track/ui/screen/TrackViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.arielwang.workoutlogger.features.track.ui.screen
+
+import androidx.lifecycle.ViewModel
+import com.arielwang.workoutlogger.features.exercise.ui.screen.ExerciseDestination
+import com.arielwang.workoutlogger.features.home.ui.screen.HomeDestination
+import com.arielwang.workoutlogger.navigate.Navigator
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import javax.inject.Inject
+
+object TrackView {
+    data class State(
+        val notKnow: String = ""
+    )
+
+    sealed class Action {
+        object GoToNextPage : Action()
+        object GoBackToPreviousPage : Action()
+    }
+}
+
+@HiltViewModel
+class TrackViewModel @Inject constructor(
+    private val navigator: Navigator
+) : ViewModel() {
+
+    private var viewState = TrackView.State()
+
+    private val _uiState = MutableStateFlow(viewState)
+    val uiState: StateFlow<TrackView.State> = _uiState
+
+    fun onUiAction(action: TrackView.Action) {
+        when (action) {
+            is TrackView.Action.GoToNextPage -> {
+                navigator.navigate(HomeDestination.route())
+            }
+            is TrackView.Action.GoBackToPreviousPage -> {
+                navigator.navigate(ExerciseDestination.route())
```

Comment:
If were want to go back, we will need to use navigator.navigateUp. If we navigate to the previous destination, this will add the destination at the top of the navigation stack. The stack would be like:
Home -> Exercise -> Track -> Exercise

Reply:
thank you so much for explaining this to me.